File: examples/developer-hub-javascript/deploy_contract_coston2.js
// THIS IS EXAMPLE CODE. DO NOT USE THIS CODE IN PRODUCTION.
import { Web3 } from "web3";
import fs from "fs";

const web3 = new Web3(
  new Web3.providers.HttpProvider(
    "https://coston2-api.flare.network/ext/C/rpc",
  ),
);

const bytecode = fs.readFileSync("./build/FtsoV2FeedConsumer.bin", "utf8");
import abi from "./build/FtsoV2FeedConsumer.json" assert { type: "json" };
const FtsoV2FeedConsumer = new web3.eth.Contract(abi);

FtsoV2FeedConsumer.handleRevert = true;

async function deploy() {
  const privateKey = process.env.ACCOUNT_PRIVATE_KEY.toString();
  const wallet = web3.eth.accounts.wallet.add(privateKey);

  const contractDeployer = FtsoV2FeedConsumer.deploy({
    data: "0x" + bytecode,
  });

  const tx = await contractDeployer.send({
    from: wallet[0].address,
    nonce: await web3.eth.getTransactionCount(wallet[0].address),
    gasPrice: await web3.eth.getGasPrice(),
  });
  console.log("Contract deployed at address: " + tx.options.address);
}

deploy();

File: examples/developer-hub-rust/src/bin/deploy_contract_coston2.rs
// THIS IS EXAMPLE CODE. DO NOT USE THIS CODE IN PRODUCTION.
use alloy::{
    network::EthereumWallet, providers::ProviderBuilder, signers::local::PrivateKeySigner, sol,
};
use eyre::Result;

sol!(
    #[sol(rpc)]
    FtsoV2FeedConsumer,
    "./FtsoV2FeedConsumer.json"
);

#[tokio::main]
async fn main() -> Result<()> {
    let private_key = std::env::var("ACCOUNT_PRIVATE_KEY")?;

    let signer: PrivateKeySigner = private_key.parse().unwrap();
    let wallet = EthereumWallet::from(signer.clone());
    let provider = ProviderBuilder::new()
        .with_recommended_fillers()
        .wallet(wallet)
        .on_http("https://coston2-api.flare.network/ext/C/rpc".parse()?);

    let contract = FtsoV2FeedConsumer::deploy(&provider).await?;
    println!("Deployed contract at address: {}", contract.address());
    Ok(())
}
